core/test/unit/middleware/cors_spec.js

```diff
@@ -137,4 +141,41 @@ describe('cors', function () {
 
         done();
     });
+
+    it('should be enabled if the origin matches config.url', function (done) {
+        var origin = 'http://my.blog';
+        configUtils.set({
+            url: origin
+        });
+
+        req.get = sinon.stub().withArgs('origin').returns(origin);
+        res.get = sinon.stub().withArgs('origin').returns(origin);
+        req.headers.origin = origin;
+
+        cors(req, res, next);
+
+        next.called.should.be.true();
+        res.headers['Access-Control-Allow-Origin'].should.equal(origin);
+
+        done();
+    });
```

Comment:
Okay, now I see what you're testing. Without that additional context in the test it seems like we're testing a same-origin request.